src/main/java/io/github/christophermanahan/carnitas/Main.java

```diff
@@ -4,12 +4,12 @@
 import java.net.ServerSocket;
 
 public class Main {
-
-    public static ServerSocket serverSocket;
-
     public static void main(String[] args) throws IOException {
         int port = args.length == 0 ? 33333 : Integer.parseInt(args[0]);
-        serverSocket = new ServerSocket(port);
-        new HTTPServer(new ServerSocketListener(serverSocket), new ErrorLogger()).run();
+        while (true) {
+            ServerSocket serverSocket = new ServerSocket(port);
+            new HTTPServer(new ServerSocketListener(serverSocket), new ErrorLogger()).run();
+            serverSocket.close();
```

Comment:
Do we need to close on each iteration?

Reply:
No, I believe this is unnecessary after pulling the ServerSocket out of the loop as stated above.